tests/scripts/all.sh

```diff
@@ -898,9 +898,12 @@ component_test_full_cmake_clang () {
 }
```

Comment:
Sorry, which commit?

Reply:
The commit message of this commit "Turn off deprecated features in component_build_deprecated."

Reply:
Ah, I guess it's the name of the component that's confusing. What it actually does is to check that if you build the library _without_ deprecated features, and you arrange disable deprecated features (in either of two ways: compiler warning + `-Werror`, or compiling them out altogether), the build still works. I guess it should be called `component_build_without_deprecated`.

Reply:
Yes I agree that component_build_without_deprecated would be a better name in that case. Probably not for this PR to change that name though but if you include some of your above explanation in the commit message it would be all clear.